From fe0306369d7bd1b2354507838b5f74944c5a78a2 Mon Sep 17 00:00:00 2001
From: Ravi Patel <ravi.patel@xilinx.com>
Date: Thu, 16 Jul 2020 22:57:23 -0700
Subject: [PATCH 1441/1852] firmware: xilinx: Use u32 for reset ID in reset
 APIs

commit 0071bce40998c35b5bb91adc4ae0f953612f92ca from
https://github.com/Xilinx/linux-xlnx.git

Instead of zynqmp_pm_reset enum, use u32 for reset ID argument
in reset APIs to extend support for versal reset IDs.

Signed-off-by: Ravi Patel <ravi.patel@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/firmware/xilinx/zynqmp.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/firmware/xilinx/zynqmp.c b/drivers/firmware/xilinx/zynqmp.c
index 1883e57d2e7e..159ec5e9e927 100644
--- a/drivers/firmware/xilinx/zynqmp.c
+++ b/drivers/firmware/xilinx/zynqmp.c
@@ -606,7 +606,7 @@ static int zynqmp_pm_ioctl(u32 node_id, u32 ioctl_id, u32 arg1, u32 arg2,
  *
  * Return: Returns status, either success or error+reason
  */
-static int zynqmp_pm_reset_assert(const enum zynqmp_pm_reset reset,
+static int zynqmp_pm_reset_assert(const u32 reset,
 				  const enum zynqmp_pm_reset_action assert_flag)
 {
 	return zynqmp_pm_invoke_fn(PM_RESET_ASSERT, reset, assert_flag,
@@ -620,8 +620,7 @@ static int zynqmp_pm_reset_assert(const enum zynqmp_pm_reset reset,
  *
  * Return: Returns status, either success or error+reason
  */
-static int zynqmp_pm_reset_get_status(const enum zynqmp_pm_reset reset,
-				      u32 *status)
+static int zynqmp_pm_reset_get_status(const u32 reset, u32 *status)
 {
 	u32 ret_payload[PAYLOAD_ARG_CNT];
 	int ret;
-- 
2.31.1

